keps/sig-instrumentation/3498-extending-stability/README.md

```diff
@@ -203,6 +209,32 @@ We will not attempt to parse metrics which:
 
 As an aside, much of this work has already been done, but is stashed in a local repo. 
 
+### Semantic of Stability Levels
+
+#### Internal Metrics
+
+`Internal` metrics have no stability guarantees and are **not** parseable by the static analysis framework. As such, `Internal` metrics will NOT be included in metric auto-documentation.
+
+#### Alpha Metrics
+
+Alpha metrics have no stability guarantees but are parseable by the static analysis framework. As such, `Alpha` metrics will be included in metric auto-documentation.
+
+#### Beta Metrics
+
+`Beta` metrics have *some* stability guarantees. Specifically, we guarantee that:
+
+ - `Beta` metrics will not be removed without first being explicitly deprecated. After deprecation, the metric will be removed in 4 months or 1 release.
+ - Furthermore, `Beta` metrics are guaranteed to be **forward compatible** in respect to alerts and queries which may be written against them. By "forward compatible", we mean that queries and alerts which are written against the metric and its labels will continue to work in the future. We ensure forward compatibility by ensuring that **labels can only be added**, *and not removed*, from `Beta` metrics.
+ - `Beta` metrics will be included in metric auto-documentation
```

Comment:
@logicalhan should we perhaps revive https://github.com/kubernetes/kubernetes/pull/85173 as a new requirement for BETA metrics? I would expect us to enforce that all BETA metrics follow the best practices

Reply:
I would defer that for GA. I don't see us making this in this release.